### DISCUSSION TITLE
Species Stats not working

### DISCUSSION THREAD

**Discussion by jflyons** [mcguirepr89/BirdNET-Pi#773] [General]
I have been running BirdNET-Pi since August 22 and greatly enjoy what it reveals about bird life around my home and garden. I hope to be more competent on the coding and all some day but for now, I can just report an issue and ask for help. I have noticed in the past day the "Species Stats" tab does not currently link to those wonderful statistical summaries.

**Comment by lloydbayley** [comment]


**Reply by ehpersonal38** [reply]
@lloydbayley @jflyons When was the last time you two did an update on your systems? I issued a commit 2 days ago (Feb 28th) which forced everyone to be on the same Streamlit version (1.11.1). I'm wondering if that broke things?

**Reply by jflyons** [reply]
I am very current on versions - showing no new updates right now. Actually I tried from a different PC/Browser setup and it slowly brought in the familiar overall view, then seems to spin away inconclusively when I retrieve specific species.

**Reply by ehpersonal38** [reply]
Hi, please perform another update and then you should be all set.

**Comment by jflyons** [comment]
Working well after this morning's six updates. Thanks!!!

**Reply by jflyons** [reply]
Fully caught up on updates, and Species Stats has decided to become difficult again - starts up with "waiting" but then comes back to main screen without going further. Help? I love BirdNET-Pi!

**Comment by tom-heisenlin** [comment]
Dear all,
after update today and reboot - Species Stats screen is empty (never had shown something).
what can I do?

**Reply by ehpersonal38** [reply]
I would recommend these actions for diagnosing this:
1. Check if `birdnetpi.local/stats` displays anything other than a blank page
2. Check the output of this command: `sudo service birdnet_stats status`. Please copy/paste this output here so I can get some more information as to what may be going awry here

**Reply by tom-heisenlin** [reply]
Hi ehpersonal38,

thank you for the reply.

some more information:
- Rapi is a 3B+
- tried to restart Tools/Services/Streamlit Statistics (active) (was and is running); restart is given no hint of what is going wrong
1. birdnetpi.local/stats brings a white empty page - even not the green one from the birdnet.pi webpage
2. sudo service birdnet_stats status:
     birdnet@birdnetpi:~/BirdNET-Pi$ sudo service birdnet_stats status
● birdnet_stats.service - BirdNET Stats
     Loaded: loaded (/home/birdnet/BirdNET-Pi/templates/birdnet_stats.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2023-05-24 07:59:40 CEST; 812ms ago
   Main PID: 761203 (streamlit)
      Tasks: 1 (limit: 779)
        CPU: 564ms
     CGroup: /system.slice/birdnet_stats.service
             └─761203 /home/birdnet/BirdNET-Pi/birdnet/bin/python3 /home/birdnet/BirdNET-Pi/birdnet/bin/streamlit run /home/birdnet/BirdNET-Pi/scripts/plotly_streamlit.py --browser.gatherUsageStats false --server.address localhost --server.baseUrlPath /stats

May 24 07:59:40 birdnetpi systemd[1]: Started BirdNET Stats.
birdnet@birdnetpi:~/BirdNET-Pi$
![BirdnetPi](https://github.com/mcguirepr89/BirdNET-Pi/assets/107929199/b6aaefed-298f-421f-8214-b505169c9c31)

**Reply by dennyChandler** [reply]
Same issue:

blank page and the output to the status is:

● birdnet_stats.service - BirdNET Stats
     Loaded: loaded (/home/denny/BirdNET-Pi/templates/birdnet_stats.service; enabled; vendor preset: enabled)
     Active: activating (start-pre) since Wed 2023-05-24 09:35:37 CDT; 10s ago
Cntrl PID: 4529 (sleep)
      Tasks: 1 (limit: 1599)
        CPU: 3ms
     CGroup: /system.slice/birdnet_stats.service
             └─4529 /bin/sleep 60

May 24 09:35:37 birdnet systemd[1]: Starting BirdNET Stats...

**Reply by dennyChandler** [reply]
Mine is cycling, it seems to restart every time I navigate to the stats page.

**Reply by ehpersonal38** [reply]
Please direct further discussion here - https://github.com/mcguirepr89/BirdNET-Pi/issues/929

**Reply by Mo-Girks** [reply]
I too had the Species Stats not update. Here are my troubleshooting steps

Initially I checked the status of the Streamlit service

`sudo service birdnet_stats status`

With the following results

> ● birdnet_stats.service - BirdNET Stats
>      Loaded: loaded (/home/girks/BirdNET-Pi/templates/birdnet_stats.service; enabled; vendor preset: enabled)
>      Active: active (running) since Fri 2023-11-10 13:09:55 AEDT; 3 days ago
>    Main PID: 2322824 (streamlit)
>       Tasks: 8 (limit: 8754)
>         CPU: 5min 42.994s
>      CGroup: /system.slice/birdnet_stats.service
>              └─2322824 /home/girks/BirdNET-Pi/birdnet/bin/python3 /home/girks/BirdNET-Pi/birdnet/bin/streamlit run /home/girks/BirdNET-Pi/scripts/plotly_streamlit.py --browser.gatherUsageStats false --server.address localhost --server.b>
> 
> Nov 10 13:09:55 birdnet systemd[1]: Started BirdNET Stats.
> Nov 10 13:10:01 birdnet streamlit[2322824]:   You can now view your Streamlit app in your browser.
> Nov 10 13:10:01 birdnet streamlit[2322824]:   URL: http://localhost:8501/stats
> Nov 10 13:10:01 birdnet streamlit[2322824]:   A new version of Streamlit is available.
> Nov 10 13:10:01 birdnet streamlit[2322824]:   See what's new at https://discuss.streamlit.io/c/announcements
> Nov 10 13:10:01 birdnet streamlit[2322824]:   Enter the following command to upgrade:
> Nov 10 13:10:01 birdnet streamlit[2322824]:   $ pip install streamlit --upgrade


While I should have just restarted the service, I instead followed the instructions from #929 , linked above.

I ran the command
`~/BirdNET-Pi/scripts/update_birdnet.sh -b streamlit-without-cache`

and had the following output

> ++ sudo -u girks git -C /home/girks/BirdNET-Pi rev-parse HEAD
> ++ set +x
> + sudo -u girks git -C /home/girks/BirdNET-Pi reset --hard
> HEAD is now at f8e74ca Update README.md
> + set +x
> + sudo -u girks git -C /home/girks/BirdNET-Pi fetch origin streamlit-without-cache
> remote: Enumerating objects: 9844, done.
> remote: Counting objects: 100% (9844/9844), done.
> remote: Compressing objects: 100% (2869/2869), done.
> remote: Total 9746 (delta 7070), reused 9509 (delta 6855), pack-reused 0
> Receiving objects: 100% (9746/9746), 106.79 MiB | 6.29 MiB/s, done.
> Resolving deltas: 100% (7070/7070), completed with 77 local objects.
> From https://github.com/mcguirepr89/BirdNET-Pi
>  * branch            streamlit-without-cache -> FETCH_HEAD
> + set +x
> + sudo -u girks git -C /home/girks/BirdNET-Pi switch -C streamlit-without-cache --track origin/streamlit-without-cache
> fatal: invalid reference: origin/streamlit-without-cache
> + set +x
> + sudo -u girks git -C /home/girks/BirdNET-Pi diff --stat f8e74cad351093f098b4e11d2fe4ef8ba6bc3482 HEAD
> + set +x
> + my_dir=/home/girks/BirdNET-Pi/scripts
> + sudo systemctl stop birdnet_server.service
> + sudo systemctl stop birdnet_recording.service
> + services=(chart_viewer.service spectrogram_viewer.service icecast2.service extraction.service birdnet_recording.service birdnet_log.service birdnet_stats.service)
> + for i in "${services[@]}"
> + sudo systemctl restart chart_viewer.service
> + for i in "${services[@]}"
> + sudo systemctl restart spectrogram_viewer.service
> + for i in "${services[@]}"
> + sudo systemctl restart icecast2.service
> + for i in "${services[@]}"
> + sudo systemctl restart extraction.service
> + for i in "${services[@]}"
> + sudo systemctl restart birdnet_recording.service
> + for i in "${services[@]}"
> + sudo systemctl restart birdnet_log.service
> + for i in "${services[@]}"
> + sudo systemctl restart birdnet_stats.service
> + sudo systemctl start birdnet_server.service
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> + sleep 5
> + for i in {1..5}
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      2551765/python3
> + logger '[/usr/local/bin/restart_services.sh] birdnet_server.service is running'
> + break
> + systemctl is-active --quiet birdnet_server.service
> + grep 5050 /dev/fd/63
> ++ netstat -tulpn
> tcp        0      0 127.0.0.1:5050          0.0.0.0:*               LISTEN      2551765/python3
> + status=0
> + ((  status != 0  ))
> + sudo systemctl restart birdnet_analysis.service

Success, the species Stats successfully updated. I will keep an eye on this non-cached version

**Comment by huminator1978** [comment]
My species stats is not working well. He stopped at 25 november. Why? Is there a solution?

![Species stats](https://github.com/user-attachments/assets/52536838-a174-4390-a171-cd33d0b9bdb3)

**Reply by huminator1978** [reply]
I did an update and now he is running again. Is that all or is there something else I should do or could do?

At least he is working 👍
